spec/models/scheduled_status_spec.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ScheduledStatus do
+  let(:account) { Fabricate(:account) }
+
+  describe 'validations' do
+    context 'when scheduled_at is less than minimum offset' do
+      subject { Fabricate.build(:scheduled_status, scheduled_at: 4.minutes.from_now, account: account) }
+
+      it 'is not valid', :aggregate_failures do
+        expect(subject).to_not be_valid
+        expect(subject.errors[:scheduled_at]).to include(I18n.t('scheduled_statuses.too_soon'))
+      end
+    end
+
+    context 'when account has reached total limit' do
+      subject { Fabricate.build(:scheduled_status, account: account) }
+
+      before do
+        allow(account.scheduled_statuses).to receive(:count).and_return(described_class::TOTAL_LIMIT)
+      end
+
+      it 'is not valid', :aggregate_failures do
+        expect(subject).to_not be_valid
+        expect(subject.errors[:base]).to include(I18n.t('scheduled_statuses.over_total_limit', limit: ScheduledStatus::TOTAL_LIMIT))
+      end
+    end
+
+    context 'when account has reached daily limit' do
+      subject { Fabricate.build(:scheduled_status, account: account, scheduled_at: Time.current.tomorrow + 10.minutes) }
+
+      before do
+        stub_const('ScheduledStatus::DAILY_LIMIT', 3)
+
+        Fabricate.times(ScheduledStatus::DAILY_LIMIT, :scheduled_status, account: account, scheduled_at: Time.current.tomorrow.beginning_of_day)
+      end
+
+      it 'is not valid', :aggregate_failures do
+        expect(subject).to_not be_valid
+        expect(subject.errors[:base]).to include(I18n.t('scheduled_statuses.over_daily_limit', limit: ScheduledStatus::DAILY_LIMIT))
+      end
+    end
```

Comment:
This sounds fine, but I'd have the base date as a common variable and/or use `travel_to` so that the test doesn't fail in the unlikely event it runs around 00:00.

Reply:
Just added `travel_to` in [e68740f](https://github.com/mastodon/mastodon/pull/30585/commits/e68740fe198c3ce5adce59bd4d1cce22f6c6ed0b).